app/storage/knowledge-document-repo.js

```diff
@@ -5,6 +5,45 @@ const { mapMetadataToBlob, mapMetadataToBase } = require('../mappers/knowledge-m
 
 const knowledgeContainer = blobServiceClient.getContainerClient(config.knowledgeContainer)
 
+const listKnowledge = async (category = '', orderBy = 'lastModified ', orderByDirection = 'Desc') => {
+  const blobs = []
+
+  const listOptions = {
+    includeCopy: false,
+    includeDeleted: false,
+    includeDeletedWithVersions: false,
+    includeLegalHold: false,
+    includeMetadata: false,
+    includeSnapshots: true,
+    includeTags: true,
+    includeUncommitedBlobs: false,
+    includeVersions: false,
+    prefix: ''
+  }
+
+  for await (const blob of knowledgeContainer.listBlobsFlat(listOptions)) {
+    const metadata = mapMetadataToBase(blob.metadata)
+
+    blob.metadata = metadata
+    if (category === '' || metadata.category === category) {
+      blobs.push(blob)
+    }
+  }
+
+  blobs.sort((a, b) => {
```

Comment:
Could maybe make this function a bit cleaner by moving out the sort into its own function.

Reply:
Refactored